### PR TITLE
Feature Documents

### PR DESCRIPTION
**Pull request by yattias**
- Adding ability to feature documents in feed (trending only)
### New options available
image

### Configuration available after clicking the settings button
image

### CODE REVIEW

File: components/Admin/AdminButton.tsx [hunk 1]
```diff
@@ -22,52 +26,123 @@ function AdminButton({
   showMessage,
 }: Args) {
 
-  const [excludeFromFeedSelectedChoices, setExcludeFromFeedSelectedChoices] = useState(["homepage", "hubs"]);
+  const [excludeFromFeedSelectedChoices, setExcludeFromFeedSelectedChoices] = useState(["homepage"]);
```
Comment (calvinhlee23) [inline components/Admin/AdminButton.tsx:29]: LOL so specific 😆

File: components/Admin/AdminButton.tsx [hunk 2]
```diff
@@ -22,52 +26,123 @@ function AdminButton({
...
-          showMessage({ show: true, error: false });
-        },
-        onError: () => {
-          setMessage("Failed to include in feed");
-          showMessage({ show: true, error: true });
-        }
-      })
+      _includeInFeed();
     }
+  },{
```
Comment (calvinhlee23) [inline components/Admin/AdminButton.tsx:127]: could you run linter on this file?. Formatting looks off.

File: components/Admin/AdminButton.tsx [hunk 3]
```diff
@@ -82,19 +157,31 @@ function AdminButton({
     }    
   }
 
+  const handleFeatureCheckbox = (id, state, event) => {
+    event.stopPropagation();
+    const isSelected = makeFeaturedChoices.includes(id);
+    if (isSelected) {
+      const newChoices = makeFeaturedChoices.filter(c => c !== id)
+      setMakeFeaturedChoices(newChoices);
```
Comment (calvinhlee23) [inline components/Admin/AdminButton.tsx:165]: newChoices is instantiated every time the function is called. No need to store as a var. Might be easier to read in 1 line
```suggestion
      setMakeFeaturedChoices(makeFeaturedChoices.filter(c => c !== id));
```

File: components/Admin/AdminButton.tsx [hunk 4]
```diff
@@ -127,13 +214,67 @@ function AdminButton({
...
+            <span className={css(styles.iconWrapper, styles.iconMakeFeature)}>{opt.icon}</span>
+            <span className={css(styles.optLabel)}>{opt.label}</span>
+            <span className={css(styles.settingsBtn)} onClick={(event) => {
+              event.stopPropagation();
+              setIsFeaturedSubmenuOpen(!isFeaturedSubmenuOpen)
+            }}>{icons.cog}</span>
+          </div>
+          <div className={css(styles.menuSettings, isFeaturedSubmenuOpen && styles.featuredMenuSettingsVisible)}>
+            <div className={css(styles.checkboxContainer)}>
+              <CheckBox
```
Comment (calvinhlee23) [inline components/Admin/AdminButton.tsx:237]: having this many @ts-ignore feels a little off. Let's consider making a new component soon! ( whoever has the bandwidth)

File: components/Admin/AdminButton.tsx [hunk 5]
```diff
@@ -160,30 +301,50 @@ function AdminButton({
...
       overrideTargetButton={styles.overrideTargetButton}
       withDownIcon={false}
-      onClose={() => setIsOpen(false)}
+      onClose={
+        () => {
+          if (isExcludeSubmenuOpen || isFeaturedSubmenuOpen) {
+            return setIsOpen(true)
+          }          
+          return setIsOpen(false)
+      }}
```
Comment (calvinhlee23) [inline components/Admin/AdminButton.tsx:310]: easier to read. Please run formatter
```suggestion
      onClose={
        () => ((isExcludeSubmenuOpen || isFeaturedSubmenuOpen) ? setIsOpen(true) : setIsOpen(false))
       }
```

**Review by calvinhlee23 — CHANGES_REQUESTED** [review]
Some comments.

File: components/Admin/AdminButton.tsx (hunk 1 above)
Comment (yattias) [reply components/Admin/AdminButton.tsx:29]: hmm good point, maybe `excludeFromFeedChoices` maybe sufficient here hehe....

File: components/Admin/AdminButton.tsx (hunk 2 above)
Comment (yattias) [reply components/Admin/AdminButton.tsx:127]: sure. I swear my linter runs but maybe not on tsx files for some reason

File: components/Admin/AdminButton.tsx (hunk 3 above)
Comment (yattias) [reply components/Admin/AdminButton.tsx:165]: 👍

File: components/Admin/AdminButton.tsx (hunk 4 above)
Comment (yattias) [reply components/Admin/AdminButton.tsx:237]: yeah, this checkbox component needs updating

**Comment by yattias** [comment]
Changes made

**Pull request merged**